climpred/classes.py

```diff
@@ -200,8 +318,76 @@ def __init__(self, xobj):
         """
 
         super().__init__(xobj)
-        self.control = {}
+        # Reserve sub-dictionary for the control simulation.
+        self._datasets.update({'control': {}})
+
+    # ----------------
+    # Helper Functions
+    # ----------------
+    def _apply_climpred_function(self, func, input_dict=None, **kwargs):
+        """Helper function to loop through references and apply an arbitrary climpred function.
+
+        Args:
+            func (function): climpred function to apply to object.
+            input_dict (dict): dictionary with the following things:
+                * ensemble: initialized or uninitialized ensemble.
+                * control: control dictionary from HindcastEnsemble.
+                * var: name of variable to target.
+                * init (bool): True if the initialized ensemble, False if uninitialized.
+        """
+        ensemble = input_dict['ensemble']
+        control = input_dict['control']
+        var = input_dict['var']
+        init = input_dict['init']
+
+        # Compute for single variable.
+        if var is not None:
+            return func(ensemble[var], control[var], **kwargs)
+        # Compute for all variables in control.
+        else:
+            init_vars, ctrl_vars = self._vars_to_drop(init=init)
+            ensemble = ensemble.drop(init_vars)
+            control = control.drop(ctrl_vars)
+            return func(ensemble, control, **kwargs)
+
+    def _vars_to_drop(self, init=True):
+        """Returns list of variables to drop when comparing
+        initialized/uninitialized to a control.
+
+        This is useful if the two products being compared do not share the same
+        variables. I.e., if the control has ['SST'] and the initialized has
+        ['SST', 'SALT'], this will return a list with ['SALT'] to be dropped
+        from the initialized.
+
+        Args:
+          init (bool, default True):
+            If `True`, check variables on the initialized.
+            If `False`, check variables on the uninitialized.
 
+        Returns:
+          Lists of variables to drop from the initialized/uninitialized
+          and control Datasets.
+        """
+        if init:
+            init_vars = [var for var in self._datasets['initialized'].data_vars]
+        else:
+            init_vars = [var for var in self._datasets['uninitialized'].data_vars]
+        ctrl_vars = [var for var in self._datasets['control'].data_vars]
+        # find what variable they have in common.
+        intersect = set(ctrl_vars).intersection(init_vars)
```

Comment:
Cool; didn't know you can do this.

Usually I do
`set(ctrl_vars) & set(init_vars)`